tasks/lammps.py

```diff
@@ -21,73 +28,149 @@ def _read_results(csv):
+        ):
+            native_speedup_errs.append(
+                native_sup
+                * sqrt(
+                    pow(native_err_single / native_single, 2)
+                    + pow(native_e / native_t, 2)
+                )
+            )
+        wasm_speedup_errs = []
+        wasm_err_single = wasm_errs["Actual"][0]
+        for wasm_sup, wasm_e, wasm_t in zip(
+            wasm_speedup, wasm_errs["Actual"], wasm_times["Actual"]
+        ):
+            wasm_speedup_errs.append(
+                wasm_sup
+                * sqrt(
+                    pow(wasm_err_single / wasm_single, 2)
+                    + pow(wasm_e / wasm_t, 2)
+                )
+            )
+
+        # Plot speed up data with error bars
+        ax[coords].errorbar(
+            wasm_times["WorldSize"],
+            wasm_speedup,
+            yerr=wasm_speedup_errs,
+            fmt=".-",
+            label="Faasm",
+            ecolor="gray",
+            elinewidth=0.8,
+            capsize=1.0,
+        )
+        ax[coords].errorbar(
+            native_times["WorldSize"],
+            native_speedup,
+            yerr=native_speedup_errs,
+            fmt=".-",
+            label="OpenMPI",
+            ecolor="gray",
+            elinewidth=0.8,
+            capsize=1.0,
+        )
+
+        if plot_elapsed_times:
+            # Plot elapsed time in a separate y axis
+            ax_et = ax[coords].twinx()
+            ax_et.errorbar(
+                wasm_times["WorldSize"],
+                wasm_times["Actual"],
+                yerr=wasm_errs["Actual"],
+                fmt=".--",
+                label="Faasm",
+                ecolor="gray",
+                elinewidth=0.8,
+                capsize=1.0,
+                alpha=0.3,
+            )
+            ax_et.errorbar(
+                native_times["WorldSize"],
+                native_times["Actual"],
+                yerr=native_errs["Actual"],
+                fmt=".--",
+                label="OpenMPI",
+                ecolor="gray",
+                elinewidth=0.8,
+                capsize=1.0,
+                alpha=0.3,
+            )
+            ax_et.set_ylabel("Elapsed time [s]")
+
+        ax[coords].title.set_text("{}-bound benchmark".format(bench))
+        ax[coords].set_ylim(bottom=0)
+        ax[coords].set_xlim(left=1)
+        ax[coords].set_xlabel("MPI World Size")
+        ax[coords].set_ylabel("Speed Up (vs 1 MPI Proc performance)")
+
+    # Print legend
+    handles, labels = ax[1].get_legend_handles_labels()
+    fig.legend(handles, labels)
+
+    fig.tight_layout()
+
+    # Manually add common axis labels
+    fig.text(0.015, 0.5, "", ha="center", va="center", rotation="vertical")
 
     if gui:
-        plt.show()
+        fig.show()
     else:
-        plt.savefig(plot_file, format=PLOTS_FORMAT)
+        fig.savefig(plot_file, format=PLOTS_FORMAT)
 
 
 @task
 def plot_resources(ctx, world_size, run=0, gui=False):
```

Comment:
Related to our offline discussion, note that I omit from this PR all other resource-related plots.